packages/api/endpoints/async-operations.js

```diff
@@ -92,26 +83,13 @@ async function del(req, res) {
     await asyncOperationPgModel.get(knex, { id });
   } catch (error) {
     if (error instanceof RecordDoesNotExist) {
-      if (!(await esAsyncOperationsClient.exists(id))) {
-        logger.info('Async Operation does not exist in Elasticsearch and PostgreSQL');
-        return res.boom.notFound('No record found');
-      }
-      logger.info('Async Operation does not exist in PostgreSQL, it only exists in Elasticsearch. Proceeding with deletion');
-    } else {
-      throw error;
+      logger.info('Async Operation does not exist PostgreSQL');
+      return res.boom.notFound('No record found');
     }
+    return res.boom.badImplementation(JSON.stringify(error));
```

Comment:
Should we `throw error` or return `res.boom.badImplementation`?
Some endpoints throws error like
https://github.com/nasa/cumulus/blob/feature/es-phase-2/packages/api/endpoints/collections.js#L191

Reply:
I think, generally, if we're catching an error and evaluating it, in endpoint code we should be able to explicitly use `boom` to throw an 'expected' error that codifies what happened rather than rely on an uncaught error, but if there's a reason we *shouldn't* catch and return from `del` (e.g. error handling upstream), we should talk about approaches generally for the endpoints.